pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -97,16 +101,75 @@ func vsphereLogin(cfg	*VSphereConfig, ctx context.Context) (*govmomi.Client, err
 }
 
 type Instances struct {
-	cfg		*VSphereConfig
+	cfg *VSphereConfig
 }
 
 // Instances returns an implementation of Instances for vSphere.
 func (vs *VSphere) Instances() (cloudprovider.Instances, bool) {
+
+	glog.V(4).Info("vsphere.Instances() called")
+
 	return &Instances{vs.cfg}, true
 }
 
+func getInstances(c *govmomi.Client, finder *find.Finder, ctx context.Context, nameFilter string) ([]string, error) {
+	//TODO: get all vms inside subfolders
+	vms, err := finder.VirtualMachineList(ctx, nameFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	var vmRef []types.ManagedObjectReference
+	for _, vm := range vms {
+		vmRef = append(vmRef, vm.Reference())
+	}
+
+	pc := property.DefaultCollector(c.Client)
+
+	var vmt []mo.VirtualMachine
+	err = pc.Retrieve(ctx, vmRef, []string{"name", "summary"}, &vmt)
+	if err != nil {
+		return nil, err
+	}
+
+	var vmList []string
+	for _, vm := range vmt {
+		vmPowerstate := vm.Summary.Runtime.PowerState
+		if vmPowerstate == "poweredOn" {
+			vmList = append(vmList, vm.Name)
+		}
+	}
+	return vmList, nil
+}
+
 func (i *Instances) List(nameFilter string) ([]string, error) {
-	return nil, nil
+	glog.V(4).Infof("vSphere List(%v) called", nameFilter)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	c, err := vsphereLogin(i.cfg, ctx)
+	if err != nil {
+		fmt.Errorf("Failed to create vSpere client: %s", err)
+	}
+	defer c.Logout(ctx)
+
+	fo := find.NewFinder(c.Client, true)
+	dc, err := fo.Datacenter(ctx, i.cfg.Global.Datacenter)
+	if err != nil {
+		glog.Warningf("Failed to find %v", err)
+		return nil, err
+	}
+
+	finderObj := fo.SetDatacenter(dc)
+	vmList, err := getInstances(c, finderObj, ctx, nameFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	glog.V(3).Infof("Found %v instances matching %v: %v",
```

Comment:
Let's be careful what we log at INFO level, this might be a DEBUG level message?